src/hyp_map.rs

```diff
@@ -209,6 +209,44 @@ impl PrivateRegion {
             }
         }
     }
+
+    // Restore private region to initial-state.
+    fn restore(&self) {
+        let mut copied = 0;
+        // We have to reset the full pages mapped for this segment.
+        let mapped_size = PageSize::Size4k.round_up(self.size as u64) as usize;
+        // Copy data at the beginning if it's present.
+        if let Some(data) = self.data {
+            // In case data is bigger than region size, write up to region end only.
+            let len = core::cmp::min(self.size, data.len());
+            let data = &data[0..len];
+            // Copy original data to umode area.
+            // Write to user mapping setting SUM in SSTATUS.
+            CSR.sstatus.modify(sstatus::sum.val(1));
+            // Safety:
+            // - this write is in a umode regions guaranteed to be mapped by HypMap in every page table.
```

Comment:
s/regions/region/

Reply:
Thanks, fixed this and the one a couple of lines below.